### PR TITLE
Added Laba 5

### PR DESCRIPTION
**Pull request by AlehFiadosau**
Added Laba 5

### CODE REVIEW

File: CarInspection/src/CarInspectionApi/Controllers/DriversController.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            }
+            catch (DateException ex)
+            {
+                _logger.LogError(ex.Message);
+
+                return BadRequest(ex.Message);
+            }
+        }
+
+        [HttpGet("updateDriver/{id}")]
```
Comment (alexey62soft) [inline CarInspection/src/CarInspectionApi/Controllers/DriversController.cs:85]: What?

File: CarInspection/src/CarInspectionApi/Controllers/DriversController.cs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            catch (NotFoundException ex)
+            {
+                _logger.LogError(ex.Message);
+                var data = new List<DriverViewModel>();
+
+                return Ok(data);
+            }
+        }
+
+        [HttpGet("getDriver/{id}")]
```
Comment (alexey62soft) [inline CarInspection/src/CarInspectionApi/Controllers/DriversController.cs:49]: Just "{id}"

File: CarInspection/src/CarInspectionApi/Controllers/DriversController.cs (hunk 1 above)
Comment (alexey62soft) [reply CarInspection/src/CarInspectionApi/Controllers/DriversController.cs:85]: Remove it.

File: CarInspection/src/CarInspectionApi/Startup.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseRouting();
+
+            app.UseAuthorization();
```
Comment (alexey62soft) [inline CarInspection/src/CarInspectionApi/Startup.cs:47]: Remove it.

File: CarInspection/src/WebCarInspection/Helpers/ApiClientHelper.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        public async Task<HttpResponseMessage> PutAsync(string requestUrl, object model)
+        {
+            var result = await _client.PutAsJsonAsync(requestUrl, model);
+
+            return result;
+        }
+
+        public async Task<HttpResponseMessage> DeleteAsync(string requestUrl)
+        {
+            var result = await _client.DeleteAsync(requestUrl);
```
Comment (alexey62soft) [inline CarInspection/src/WebCarInspection/Helpers/ApiClientHelper.cs:47]: just return await _client.DeleteAsync(requestUrl);

File: CarInspection/src/WebCarInspection/Helpers/ApiClientHelper.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+using WebCarInspection.ServerInfo;
+
+namespace WebCarInspection.Helpers
+{
+    public class ApiClientHelper : IApiClientHelper
+    {
+        private readonly HttpClient _client;
+        private readonly IOptions<ApiServerInfo> _apiServerInfo;
+
+        public ApiClientHelper(HttpClient client,
```
Comment (alexey62soft) [inline CarInspection/src/WebCarInspection/Helpers/ApiClientHelper.cs:15]: Register HttpClient in startup with baseUrl and use it here

File: CarInspection/src/WebCarInspection/WebCarInspection.csproj [hunk 6]
```diff
@@ -5,13 +5,14 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNet.WebApi.Client" Version="5.2.7" />
```
Comment (alexey62soft) [inline CarInspection/src/WebCarInspection/WebCarInspection.csproj:8]: This package useless

File: CarInspection/src/WebCarInspection/Helpers/ApiClientHelper.cs (hunk 5 above)
Comment (alexey62soft) [reply CarInspection/src/WebCarInspection/Helpers/ApiClientHelper.cs:15]: https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests

File: CarInspection/src/WebCarInspection/WebCarInspection.csproj (hunk 6 above)
Comment (AlehFiadosau) [reply CarInspection/src/WebCarInspection/WebCarInspection.csproj:8]: This package is used to transfer and send objects to json API

Comment (alexey62soft) [reply CarInspection/src/WebCarInspection/WebCarInspection.csproj:8]: Anyway, you should use clean HttpClient, or RestEase, for example.

**Pull request merged**